src/components/storage/InstallationMethod.jsx

```diff
@@ -167,7 +191,9 @@ const usePageInit = () => {
     const { appliedPartitioning, partitioning } = useContext(StorageContext);
     // If the last partitioning applied was from the cockpit storage integration
     // we should not reset it, as this option does apply the partitioning onNext
-    const needsReset = partitioning.storageScenarioId !== "use-configured-storage" && appliedPartitioning;
+    const needsReset = partitioning.storageScenarioId !== "use-configured-storage" &&
+        partitioning.storageScenarioId !== "home-reuse" &&
```

Comment:
I think, all scenarios but for the 'use-configure-storage' should reset in the installation method page enter.

Reply:
As I write above, this is the only solution I was able to come so far with for https://github.com/rhinstaller/anaconda-webui/pull/424#issuecomment-2370817524 (the commit @rvykydal
[home reuse: FIXUP? do not reset home reuse after applying](https://github.com/rhinstaller/anaconda-webui/pull/424/commits/497b94b1a3040541969e9cda96794d429cd19a47))

Reply:
I think you need to change that to run this code only on component mount:
This diff would fix this I believe

``` diff
diff --git a/src/components/storage/InstallationMethod.jsx b/src/components/storage/InstallationMethod.jsx
index 617892cb..c6abca11 100644
--- a/src/components/storage/InstallationMethod.jsx
+++ b/src/components/storage/InstallationMethod.jsx
@@ -189,14 +189,16 @@ const InstallationMethodFooterHelper = () => {
 const usePageInit = () => {
     const { setIsFormDisabled } = useContext(FooterContext);
     const { appliedPartitioning, partitioning } = useContext(StorageContext);
+    const pageHasMounted = useRef(false);
+    // Always reset the partitioning when entering the installation destination page
     // If the last partitioning applied was from the cockpit storage integration
     // we should not reset it, as this option does apply the partitioning onNext
     const needsReset = partitioning.storageScenarioId !== "use-configured-storage" &&
-        partitioning.storageScenarioId !== "home-reuse" &&
-        appliedPartitioning;
+        appliedPartitioning &&
+        pageHasMounted.current === true;
 
     useEffect(() => {
-        // Always reset the partitioning when entering the installation destination page
+        pageHasMounted.current = true;
         if (needsReset) {
             resetPartitioning();
         } else {

```

Reply:
> ```diff
> pageHasMounted.current === true;
> ```

It does (inverting this condition). Thank you!